src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -709,7 +710,7 @@ public bool TryAppendEntireSegmentFromSmuggler(DocumentsOperationContext context
             // if this segment isn't overlap with any other we can put it directly
             using (var holder = new TimeSeriesSegmentHolder(this, context, documentId, name, collectionName, fromReplicationChangeVector: changeVector, timeStamp: baseline))
             {
-                if (holder.LoadCurrentSegment())
+                if (holder.LoadClosestSegment())
```

Comment:
why? you can get here a segment with which you aren't overlapping, so it should be okay to put directly no?

Reply:
Here's one problematic scenario we have by using `LoadCurrentSegment `instead of `LoadClosestSegment`:
Two nodes A & B with mutual replication. 
Initial state on both nodes for TS "**Heartrates**":

**_Segment 1_**
```
{12:00; 70; "watches/fitbit"}
{12:01; 69; "watches/fitbit"}
.
.
.
{13:00; 74; "watches/fitbit"}
```

Node A deleted locally all entries except 12:00.
Node B deleted timestamp 12:00 and sent the updated segment to node A (before node A sent its changes to B).
Node A gets the below segment:

**_Segment 1 [Updated]_**
```
{12:01; 69; "watches/fitbit"}
.
.
.
{13:00; 74; "watches/fitbit"}
```

We are trying to read by key from the TS table so we get **`false`** all over the way and we will append the new segment from B, although it's wrong in that case. We will lose all the deleted values and overwrite them with live ones. which can explain why we are not getting sometimes an empty TS.   
Also, we are not checking the conflict status in that case although it is conflicted...

Reply:
but this is a conflict and we should check it.. so we will go to a different code path of read & merging the values.

Reply:
the caller should check it, no?

Reply:
> but this is a conflict and we should check it.. so we will go to a different code path of read & merging the values.

Exactly! but in the scenario I described, we are not checking the CVs because we don't have an "appropriate" segment...

Reply:
why we don't have? if we don't then we should fail on the overlap check

Reply:
We are checking the conflict status only if we get true from `ReadByKey `(in `TryAppendEntireSegment`, line 665).. but we don't. then, we'll also return false from `IsOverlapping `because `prevLastTimestamp (=12:00) >= baseline (=12:01)` is false.

Reply:
`LoadCurrentSegment` == `ReadByKey`
I don't see how we can get `false` at `ReadByKey` but `true` on `LoadCurrentSegment`

Reply:
I think B will send TimeSeriesDeletedRangeItem and not TimeSeriesReplicationItem so we will go to a different code path (DeleteTimestampRange and not  TryPutSegmentDirectly)

Reply:
> `LoadCurrentSegment` == `ReadByKey` I don't see how we can get `false` at `ReadByKey` but `true` on `LoadCurrentSegment`

you are right, and there is why I changed it to `LoadClosestSegment`.

Reply:
@shiranshalom Can you check the following case:

local segments:
```
{11:00; 70; "watches/fitbit"}
{11:10; 69; "watches/fitbit"}
.
.
.
{11:50; 74; "watches/fitbit"}
```
```
{12:00; 70; "watches/fitbit"}
{12:10; 69; "watches/fitbit"}
.
.
.
{12:50; 74; "watches/fitbit"}
```
```
{13:00; 70; "watches/fitbit"}
{13:10; 69; "watches/fitbit"}
.
.
.
{13:50; 74; "watches/fitbit"}
```

Incoming segment from replication:
```
{12:20; 70; "watches/fitbit"}
{12:30; 69; "watches/fitbit"}
{12:40; 74; "watches/fitbit"}
```

Can it be that we put this directly while we should iterate over each value individually?